plugins/autofarm.cpp

```diff
@@ -201,11 +201,26 @@ class AutoFarm {
     {
         // this algorithm attempts to change as few farms as possible, while ensuring that
         // the number of farms planting each eligible plant is "as equal as possible"
-
-        if (farms.empty() || plants.empty())
-            return; // do nothing if there are no farms or no plantable plants
-
+        
         int season = *df::global::cur_season;
+        
+        if (farms.empty() || plants.empty())
+        {
+            // if no more plants were requested, fallow all farms
+            // if there were no farms, do nothing
+            for (auto farm : farms)
+            {
+                int o = farm->plant_id[season];
+                if (o != -1)
+                {
+                    farm->plant_id[season] = -1;
+                    out << "autofarm: changing farm #" << farm->id <<
+                        " from " << ((o == -1) ? "NONE" : world->raws.plants.all[o]->name) <<
```

Comment:
A more "DFHack-y" way would be to use `df::plant_raw::find(o)` here, which would give you a `plant_raw*`, set to null if the ID is invalid (like -1). It wouldn't quite be a drop-in replacement (we don't have an idiomatic way to access a field of a possibly-null pointer that I know of), but it would look something like this:
```c++
df::plant_raw *raw = df::plant_raw::find(o);
// ...
out << (raw ? raw->name : "NONE")
```

Reply:
Very well!

Since similar code is used elsewhere as well (to set the farms when new plants _are_ requested), I have opted to factor out the simple functions `get_plant_name` and `set_farm`.

Reply:
Cool, thanks!